src/server/scripts/Outland/zone_netherstorm.cpp

```diff
@@ -576,11 +582,565 @@ class spell_netherstorm_socrethars_stone : public SpellScript
+    uint32 total;
+    bool followLeaderCalled;
+    static ObjectGuid talkingFollowerGuid;
+
+    void Reset() override
+    {
+        followLeaderCalled = false;
+        me->setActive(true);
+        me->SetFaction(me->GetCreatureTemplate()->faction);
+        me->SetReactState(REACT_PASSIVE);
+    }
+
+    void SetLeaderGuid(ObjectGuid guid, uint32 i, uint32 t)
+    {
+        leaderGuid = guid;
+        index = i;
+        total = t;
+
+        if (Creature* leader = ObjectAccessor::GetCreature(*me, leaderGuid))
+            me->SetHomePosition(leader->GetPosition());
+
+        if (me->GetEntry() == NPC_SAEED_FOLLOWER_1 && talkingFollowerGuid.IsEmpty())
+            talkingFollowerGuid = me->GetGUID();
+    }
+
+    void DoAction(int32 action) override
+    {
+        if (action == 1)
+        {
+            me->SetFaction(250);
+            me->SetReactState(REACT_AGGRESSIVE);
+            FollowLeader();
+        }
+    }
+
+    void FollowLeader(Unit* who = nullptr)
+    {
+        if (Creature* leader = ObjectAccessor::GetCreature(*me, leaderGuid))
+        {
+            if (!who)
+            {
+                float radius = 3.0f;
+                float leaderOri = leader->GetOrientation();
+                float angle = 2.0f * M_PI * float(index) / float(total);
+                float finalAngle = leaderOri + angle;
+
+                me->GetMotionMaster()->Clear();
+                me->GetMotionMaster()->MoveFollow(leader, radius, finalAngle);
+                me->SetHomePosition(me->GetPositionX(), me->GetPositionY(), me->GetPositionZ(), me->GetOrientation());
+            }
+            else
+            {
+                me->AI()->AttackStart(who);
+            }
+        }
+    }
+
+    void UpdateAI(uint32 /*diff*/) override
+    {
+        if (leaderGuid.IsEmpty())
+            return;
+
+        Creature* leader = ObjectAccessor::GetCreature(*me, leaderGuid);
+        if (!leader || !leader->IsAlive())
+        {
+            me->DespawnOrUnsummon(1s, 5s);
+            return;
+        }
+
+        me->SetHomePosition(leader->GetHomePosition());
+
+        if (leader->GetVictim())
+        {
+            if (!UpdateVictim())
+                AttackStart(leader->GetVictim());
+
+            DoMeleeAttackIfReady();
+            followLeaderCalled = false;
+        }
+        else
+        {
+            if (!followLeaderCalled)
+            {
+                FollowLeader();
+                followLeaderCalled = true;
+            }
+        }
+    }
+};
+
+// Initialisiere die statische Variable
+ObjectGuid npc_saeed_follower::talkingFollowerGuid;
+
+CreatureAI* GetAI_npc_saeed_follower(Creature* creature)
+{
+    return new npc_saeed_follower(creature);
+}
+
+
+struct npc_captain_saeed : public EscortAI
+{
+    npc_captain_saeed(Creature* creature) : EscortAI(creature), playerAwayTime(0), escortPaused(false), pausedByPlayer(false), pausedByEvent(false), attackScheduled(false) { }
+
+    std::list<Creature*> followers;
```

Comment:
all of these should be under private: and using _followers as naming, _followerGUIDs, etc.